flink-core/src/main/java/org/apache/flink/core/plugin/PluginUtils.java

```diff
@@ -32,22 +29,24 @@
  */
 public final class PluginUtils {
 
-	private static final Logger log = LoggerFactory.getLogger(PluginUtils.class);
-
 	private PluginUtils() {
 		throw new AssertionError("Singleton class.");
 	}
 
 	public static void initPluginManagerSingletonFromRootFolder(Optional<Path> pluginsRootPath) {
-		Collection<PluginDescriptor> pluginDescriptorsForDirectory = Collections.emptyList();
+		Collection<PluginDescriptor> pluginDescriptorsForDirectory;
+
 		if (pluginsRootPath.isPresent()) {
 			try {
 				pluginDescriptorsForDirectory =
 					new DirectoryBasedPluginFinder(pluginsRootPath.get()).findPlugins();
-			} catch (IOException ioex) {
-				log.warn("Exception when locating plugins in root folder {}. Ignoring plugins.", pluginsRootPath, ioex);
+			} catch (IOException e) {
+				throw new IllegalStateException("Exception when trying to initialize plugin system.", e);
```

Comment:
`IllegalStateException`? Maybe `FlinkException` or it's subclass like `ConfigurationException`? I'm not sure if this qualifies as "illegal state"?